src/repo/mod.rs

```diff
@@ -235,8 +238,34 @@ impl TryFrom<RequestKind> for RepoEvent {
     }
 }
 
+#[derive(Debug)]
+struct Lock {
+    file: File,
+}
+
+impl Lock {
+    fn new(path: &Path, create: bool) -> std::io::Result<Lock> {
+        use std::fs::OpenOptions;
+
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(create)
+            .open(path)
+            .unwrap();
+
+        FileExt::try_lock_exclusive(&file)?;
```

Comment:
I guess this way of invocation is ok as well, perhaps more usual way to call would be  `file.try_lock_exclusive()?`.